src/common/cockpittest.c

```diff
@@ -481,8 +481,15 @@ _cockpit_assert_data_eq_msg (const char *domain,
     len = strlen (data);
   if (exp_len < 0)
     exp_len = strlen (expect);
-  if (len == exp_len && memcmp (data, expect, len) == 0)
-    return;
+
+  if (len == exp_len)
+    {
+      if (len == 0)
+        return;
+      else if (data && expect && memcmp (data, expect, len) == 0)
+        return;
+    }
+
```

Comment:
I think the `&&` is correct, since we want to print a message if the data isn't equal. As you said on IRC, we probably should check for `NULL` before using `strlen` though.